Adobe DTM Driver

briangonzalez opened the pull request.

poteto committed:
Merge branch 'release/0.5.3' into develop

Commit:
Automatically remove all unused adapters from build

review-ninja commented:
<a href="https://app.review.ninja/dollarshaveclub/ember-metrics/pull/2" target="_blank"><img src="https://app.review.ninja/50868894/pull/2/badge" alt="ReviewNinja"/>

briangonzalez commented on addon/metrics-adapters/google-analytics.js:
```diff
@@ -20,6 +20,7 @@ export default BaseAdapter.extend({
   },
 
   init() {
+    console.log('Hello world')
```
!fix remove this `console.log`

briangonzalez replied:
!resolved

briangonzalez commented:
+1

briangonzalez commented:
-1

briangonzalez commented:
:+1:

homu committed:
Auto merge of #56 - jasonmit:cherrypick-adapters, r=poteto

Tree shake unused adapters

This PR addresses #24 so adapters referenced in `config.metricsAdapters` and `config['ember-metrics'].includeAdapters` are bundled into the application.  Everything other adapter, with the exception to base, is excluded from the build.

## Note
`config['ember-metrics'].includeAdapters` was added to continue to support the lazy loading scenario w/o having to incur the cost of having to include all adapters.

I'm open to bike shed.  Thanks again for the addon!

Fixes #24

poteto committed:
Update ember-cli

homu committed:
Auto merge of #60 - poteto:update-ember-cli, r=poteto

Update ember-cli

sly7-7 committed:
Add Piwik Adapter

sly7-7 committed:
update ember cli to 2.3.0

homu committed:
Auto merge of #63 - sly7-7:update-ember-cli, r=poteto

update ember cli to 2.3.0

Closes #61

homu committed:
Auto merge of #62 - sly7-7:piwik-adapter, r=poteto

Add Piwik Adapter

Closes #58

homu committed:
Auto merge of #64 - poteto:develop, r=poteto

Sync master with develop

andrewmartin committed:
initial setup and some configuration for the dtm driver

setting up some intial DTM methods.

- satellite calls are flexible
- added satellite.pageBottom() but may want to consider adding to routing

improving driver

- removing redundant callSatellite method
- adding proper trackEvent naming with optional `label` flag
  - presently, the naming follows the format `category - action - label`
  - if `label` isn't present, the event name is simply `category - action`

fixing adobe dtm driver to avoid throwing undefined errors

- reorganization of pageView and separation of concerns to a couple new methods

changing conditional

major updates to metrics driver for event queueing, config

- when _satellite is not defined, push events to array
	- an interval runs checking for this array length
	- this interval pushes all events in array
- option to configure data layer name on window object
  - falls back to 'dtmDataLayer'
  - must be set in Adobe DTM under 'Data Elements'
  - once set, _satellite.getVar('keyName') works within DTM
- some general cleanup and removal of unused variables

adding page bottom queue handler

adding pushPayload method

removing pageBottom methods and queue handler;
rename "listen" methods to use the word "check"

achambers committed:
Implement 'trackPage' in MixPanel adapter

As Mixpanel is mostly centered around Events as opposed to Page Views,
it doesn't strictly have an API for tracking page views. This is the
reason that the trackPage function was not initially implemented. However, it is
useful to track page views as events.

The implementation for the trackPage function is as suggested by the
[MixPanel
docs](https://mixpanel.com/help/questions/articles/why-measure-events-vs-page-views)

homu committed:
Auto merge of #68 - achambers:implement-mixpanel-track-page, r=poteto

Implement 'trackPage' in MixPanel adapter

As Mixpanel is mostly centred around Events as opposed to Page Views,
it doesn't strictly have an API for tracking page views. This is the
reason that the trackPage function was not initially implemented. However, it is
useful to track page views as events.

The implementation for the trackPage function is as suggested by the
[MixPanel docs](https://mixpanel.com/help/questions/articles/why-measure-events-vs-page-views)

poteto committed:
Released 0.6.0

jrdn91 committed:
Update to identify method for mixpanel adapter

Mixpanels API specifies that identify only sets a new user up with the passed in ID, so other properties about the user are not passed in.
https://mixpanel.com/help/reference/javascript-full-api-reference#mixpanel.identify
Instead it should pair `identify` with `people.set` to assign more details about the user to that user id
https://mixpanel.com/help/reference/javascript-full-api-reference#mixpanel.people.set

Another method should probably be included to just set the user information if it changes as once you've identified or aliased a user with the active session, updating the user with `people.set` should just use the previously identified or aliased user to update with the info passed into `people.set`

sly7-7 committed:
Document invoke as a public api

homu committed:
Auto merge of #72 - sly7-7:make-invoke-public, r=poteto

Document invoke as a public api

Closes #66

Commit:
Use `Ember.assign()` if possible

homu committed:
Auto merge of #78 - Cryrivers:master, r=poteto

Use `Ember.assign()` if possible

As `Ember.merge` is deprecated since `Ember 2.5.0`.

poteto committed:
Released 0.6.1

mpeven committed:
add in tests for people.set

homu committed:
Auto merge of #80 - mpeven:master, r=poteto

Tests for identify/people.set

I added in tests for #70.

@poteto: I've been using Ember for a while now, but this is my first contribution ever to any open source project. Please let me know if this is good or if I should do anything differently.

poteto committed:
Released 0.6.2
